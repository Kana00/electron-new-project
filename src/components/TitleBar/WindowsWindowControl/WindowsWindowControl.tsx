import * as React from 'react';
import * as electron from 'electron';
import Flexbox from 'flexbox-react';
import ButtonThemify from '../../ButtonThemify/ButtonThemify';

export default class WindowsWindowControl extends React.Component<WindowsWindowControlPropsType, WindowsWindowControlStateType> {
  private sizePixel: number | string;
  private electronWindow: Electron.BrowserWindow;
  constructor(props: WindowsWindowControlPropsType) {
    super(props);

    this.electronWindow = electron.remote.getCurrentWindow();

    this.state = {
      isMaxSizeWindow: false,
      isOverMinus: false,
      isOverPlus: false,
      isOverClose: false
    };

    this.maximizeWindow = this.maximizeWindow.bind(this);

  }

  maximizeWindow() {
    if (this.state.isMaxSizeWindow === false) {
      if (!this.electronWindow.isMaximized()) {
        this.setState({ isMaxSizeWindow: true });
        this.electronWindow.maximize();
      }
    } else {
      if (this.electronWindow.isMaximized()) {
        this.setState({ isMaxSizeWindow: false });
        this.electronWindow.unmaximize();
      }
    }
  }

  render() {
    this.sizePixel = this.props.height.toString();
    this.sizePixel.substring(0, this.sizePixel.length - 2);
    try {
      this.sizePixel = parseInt(this.sizePixel);
      this.sizePixel = this.sizePixel / 2;
      this.sizePixel = this.sizePixel + 'px';
    } catch (e) { console.log(e); }

    const styles = {
      controlContainer: {
        color: this.props.colorOfControl,
        lineHeight: (this.props.height) ? this.props.height : '100%',
        fontSize: (this.props.height) ? this.sizePixel : '100%',
      }
    };
    return (
      <Flexbox flexDirection='row' justifyContent='center' alignItems='center' style={styles.controlContainer}>

        <ButtonThemify textColor={this.props.colorOfControl} codeThemify='ti-minus' click={() => this.electronWindow.minimize()}/>
        {
          (this.state.isMaxSizeWindow) ?
          <ButtonThemify textColor={this.props.colorOfControl} codeThemify='ti-layers' click={() => this.maximizeWindow()}/>
          :
          <ButtonThemify textColor={this.props.colorOfControl} codeThemify='ti-control-stop' click={() => this.maximizeWindow()}/>
        }
        <ButtonThemify textColor={this.props.colorOfControl} codeThemify='ti-close' click={() => this.electronWindow.close()}/>

      </Flexbox>
    );
  }
}
