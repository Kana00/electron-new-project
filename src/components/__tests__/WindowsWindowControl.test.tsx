import * as React from 'react';
import WindowsWindowControl from '../TitleBar/WindowsWindowControl/WindowsWindowControl';
import * as renderer from 'react-test-renderer';

describe('Test component WindowsWindowControl', () => {
  it('render correctly', () => {
    const tree = renderer.create(
    <WindowsWindowControl
    colorOfControl='#BBBBBB'
    onOverColor='#EEEEEE'
    height='30px'/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
