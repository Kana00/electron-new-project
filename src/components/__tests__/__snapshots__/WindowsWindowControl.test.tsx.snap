// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component WindowsWindowControl render correctly 1`] = `
<div
  className="sc-bdVaJa dkRebT"
  style={
    Object {
      "color": "#BBBBBB",
      "fontSize": "15px",
      "lineHeight": "30px",
    }
  }
>
  <div
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <span
      className="ti-minus"
    />
  </div>
  <div
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <span
      className="ti-plus"
    />
  </div>
  <div
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    <span
      className="ti-close"
    />
  </div>
</div>
`;
